src/components/common/mixins/forms/types.ts

```diff
@@ -0,0 +1,160 @@
+import type { Validator } from '../../validators.js';
+
+export type FormRestoreMode = 'autocomplete' | 'restore';
+export type FormValueType = string | File | FormData | null;
+
+declare class BaseFormAssociatedElement {
+  public static readonly formAssociated: boolean;
+
+  private __internals: ElementInternals;
```

Comment:
PS: This is probably redundant since it's private and can't reflect on classes using the mixin in any way.
Still a bit iffy on the declared classes since they can't enforce the implementation, but I get why we need them.